src/metabase/public_settings.clj

```diff
@@ -468,3 +468,16 @@
   :type       :json
   :setter     :none
   :getter     fetch-cloud-gateway-ips-fn)
+
+(defsetting enable-database-syncing-modal
+  (str (deferred-tru "Whether an introductory modal should be shown after the next database connection is added.")
+       " "
+       (deferred-tru "Defaults to false if any non-default database connections already exist for this instance."))
+  :visibility :admin
+  :type       :boolean
+  :getter     (fn []
+                (let [v (setting/get-boolean :enable-database-syncing-modal)]
+                  (if (nil? v)
+                    (not (db/exists? 'Database :is_sample false))
+                    ;; frontend should set this value to `true` after the modal has been shown once
+                    v))))
```

Comment:
Had to make this a getter since I was getting a cyclic dependency error if I just tried making the default value `(not (db/exists? 'Database :is_sample false))`

Reply:
what dependency was a problem? That sounds like a strange error.

Reply:
Dependency between `metabase.public-settings` and `metabase.models.database`, via `metabase.api.common`. Quoting `Database` in the read doesn't resolve it. Maybe something to do with how or when `defsetting` resolves default values?